### PR TITLE
 Network restore case

### PR DESCRIPTION
**Pull request by nikhil-achole**

### CODE REVIEW

**Comment by maheshbhoge** [comment]
Somehow we need to ensure that , this testcase should run as last case of functinal suite . otherwise it will hamper next testcases to be run as we are deleting all private networks.

File: tempest/api/workloadmgr/restore/test_network_restore.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+    credentials = ['primary']
+
+    @classmethod
+    def setup_clients(cls):
+        super(WorkloadTest, cls).setup_clients()
+        cls.client = cls.os.wlm_client
+        reporting.add_test_script(str(__name__))
+
+    def delete_network_topology(self):
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_network_restore.py]: This should be part of base.py

File: tempest/api/workloadmgr/restore/test_network_restore.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        LOG.debug("Deleting the existing networks")
+        networkslist = self.networks_client.list_networks()['networks']
+
+        for network in networkslist:
+            if network['router:external'] == False:
+                self.delete_network(network['id'])
+            else:
+                pass
+
+    def create_network(self):
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_network_restore.py]: Include this in base.py as its common method to create network

File: tempest/api/workloadmgr/restore/test_network_restore.py [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        self.network_client.add_router_interface_with_port_id(routers['Router-2'], portid1)
+        portid2 = self.network_client.create_port(**{'network_id':nets['Private-2'], 'fixed_ips': [{'ip_address':'10.10.2.5'}]})['port']['id']
+        portid3 = self.network_client.create_port(**{'network_id':nets['Private-2'], 'fixed_ips': [{'ip_address':'10.10.2.6'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-4'], portid2)
+        self.network_client.add_router_interface_with_port_id(routers['Router-5'], portid3)
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-4'], subnets['PS-5'])
+        portid4 = self.network_client.create_port(**{'network_id':nets['Private-5'], 'fixed_ips': [{'ip_address':'10.10.5.3'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-5'], portid4)
+        
+    def get_topology_details(self):
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_network_restore.py]: Move to base.py

File: tempest/api/workloadmgr/restore/test_network_restore.py [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            nws = [x['id'] for x in networkslist]
+            nt= [{str(i):str(j) for i,j in x.items() if i not in ('network_id', 'subnets', 'created_at', 'updated_at', 'id')} for x in networkslist]
+            networks = {}
+            for each in nt:
+                networks[each['name']] = each
+
+            sbnt = self.subnets_client.list_subnets()['subnets']
+            sbnts = [{str(i):str(j) for i,j in x.items() if i not in ('network_id', 'created_at', 'updated_at', 'id')} for x in sbnt]
+            subnets = {}
+            for each in sbnts:
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_network_restore.py]: Please use variable names which convey the details appropriately. Rename this variable 'each' .

File: tempest/api/workloadmgr/restore/test_network_restore.py [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        self.network_client.add_router_interface_with_subnet_id(routers['Router-4'], subnets['PS-5'])
+        portid4 = self.network_client.create_port(**{'network_id':nets['Private-5'], 'fixed_ips': [{'ip_address':'10.10.5.3'}]})['port']['id']
+        self.network_client.add_router_interface_with_port_id(routers['Router-5'], portid4)
+        
+    def get_topology_details(self):
+            networkslist = self.networks_client.list_networks()['networks']
+            nws = [x['id'] for x in networkslist]
+            nt= [{str(i):str(j) for i,j in x.items() if i not in ('network_id', 'subnets', 'created_at', 'updated_at', 'id')} for x in networkslist]
+            networks = {}
+            for each in nt:
```
Comment (savitha-p) [inline tempest/api/workloadmgr/restore/test_network_restore.py]: Rename 'each' to network for better readability

**Pull request merged**